test/remote/gateways/remote_priority_test.rb

```diff
@@ -147,6 +162,22 @@ def test_successful_purchase_with_additional_options
     assert_equal 'Approved or completed successfully', response.message
   end
 
+  def test_successful_credit
+    options = @options.merge(@additional_creditoptions)
+    response = @gateway.credit(@credit_amount, @credit_card, options)
+    assert_success response
+    assert_equal 'Approved or completed successfully', response.message
+  end
+
+  def test_failed_credit_missing_card_month
+    card_without_month = credit_card('4242424242424242', month: '')
```

Comment:
Is this the only circumstance that would result in a failed credit? For example, I tried editing the test after pulling down your branch and passed in a nil for the amount and a valid card with expiration date and it allowed the transaction to be successful. That might be something you should validate on your end.